src/main/java/org/broadinstitute/hellbender/tools/dragstr/CalibrateDragstrModel.java

```diff
@@ -473,27 +480,120 @@ private StratifiedDragstrLocusCases collectCaseStatsParallel(final List<SimpleIn
+                                    final int resultSize = result.size();
+                                    synchronized (numberBasesProcessed) {
+                                        final long processed = numberBasesProcessed.updateAndGet(l -> l + shard.size());
+                                        progressMeter.update(absoluteCoordinates.toSimpleInterval(processed, 1), resultSize);
                                     }
-                                }).reduce(StratifiedDragstrLocusCases::merge).orElseGet(() -> new StratifiedDragstrLocusCases(hyperParameters.maxPeriod, hyperParameters.maxRepeatLength)));
+                                    return result;
+                                } catch (final IOException ex) {
+                                    throw new GATKException("problems accessing the str-table-file contents at " + strTablePath, ex);
+                                }
+                            })
+                            .reduce(StratifiedDragstrLocusCases::merge)
+                            .orElseGet(() -> new StratifiedDragstrLocusCases(
+                                    hyperParameters.maxPeriod,
+                                    hyperParameters.maxRepeatLength)));
+        }
+    }
+
+    // Spliterator that splits by segragating even and odd elements as supposed to
+    // first and second half which the standard (e.g {@link Collection#stream} delgates onto {@link ArraySpliterator}).
+    //
+    // This approach makes the load more balanced between threads and the excecution order follows more closelly
+    // the input source list order.
+    //
+    // For its single use in this tool, this makes it more likely that there won't be a need to hold
+    // the full genome in memory by the underlaying ReferenceSource and that the
+    // actual progress corresponds more closely to the one reported by the progress-meter (always following
+    // sreference coordinate order).
+    private static class InterleavingListSpliterator<T> implements Spliterator<T> {
+
+        private final List<T> source;
+        private int next;
+        private int increment;
+        private int remaining;
+
+        private InterleavingListSpliterator(final List<T> source) {
+            this(source, 0, 1);
+        }
+
+        private InterleavingListSpliterator(final List<T> source, final int start, final int increment) {
+            Utils.nonNull(this.source = source);
+            final int sourceSize;
+            this.next = Utils.validIndex(start, sourceSize = this.source.size());
+            Utils.validate((this.increment = increment) > 0, "increment must be extrictly positive");
+            // we add increment -1 to numerator to get the ceiling of the int div.
+            this.remaining = (sourceSize - start + increment - 1) / increment;
+        }
+
+        @Override
+        public boolean tryAdvance(Consumer<? super T> action) {
```

Comment:
Given the math this spliterator is operating with I would probably request at least a few tests that this works reasonably well given all the strange edge cases. I would suggest building a unit test that generates lists from size 1 -> n and then runs these splitorators with a number of splits 1 -> x and asserts that in every case every element is necessarily getting hit.

Reply:
One of the new commits adds that testing.